Essentials/src/com/earth2me/essentials/User.java

```diff
@@ -111,13 +113,29 @@ public boolean isPermissionSet(final String node) {
     }
 
     private boolean isAuthorizedCheck(final String node) {
-
         if (base instanceof OfflinePlayer) {
             return false;
         }
 
+        // Track player permissions change to avoid calling Player#hasPermission multiple times
+        if (!lastTrackedPermission.isEmpty()) {
+            for (PermissionAttachmentInfo info : base.getEffectivePermissions()) {
+                if (lastTrackedPermission.containsKey(info.getPermission())) {
+                    if (info.getValue() != lastTrackedPermission.get(info.getPermission())) {
+                        break;
+                    }
+
+                    return true;
+                }
+            }
+            lastTrackedPermission.clear();
+        }
+
+        final boolean hasPermission = ess.getPermissionsHandler().hasPermission(base, node);
+        lastTrackedPermission.put(node, hasPermission);
+
         try {
-            return ess.getPermissionsHandler().hasPermission(base, node);
+            return hasPermission;
```

Comment:
the operation that is supposed to be in the try catch is `ess.getPermissionsHandler()#hasPermission()`